lib/api/core/httpRouter/routeHandler.js

```diff
@@ -48,10 +52,11 @@ class RouteHandler {
         this.data.jwt = headers[k].substring('Bearer '.length);
       }
       else if (k.toLowerCase() === 'x-kuzzle-volatile') {
-        this.data.volatile = JSON.parse(headers[k]);
-      }
-      else {
-        this.data[k] = headers[k];
+        try {
+          this.data.volatile = JSON.parse(headers[k]);
+        } catch (e) {
+          throw new BadRequestError('Unable to convert HTTP x-kuzzle-volatile header to JSON');
```

Comment:
nitpicking: jsdoc should now state that this constructor throws, using `@throws <reason>` (see http://usejsdoc.org/tags-throws.html)